sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/FieldShouldBeReadonly.cs

```diff
@@ -171,4 +171,15 @@ public void Do()
             var x = teststring;
         }
     }
+
+    [Serializable]
+    public class SerializableClass
+    {
+        private int field; // Compliant, containing class is marked with [Serializable]
+
+        public SerializableClass()
+        {
+            field = 5;
+        }
+    }
```

Comment:
Corner case that would be trivial to add a test for: [Serializable] is not inherited so the rule should raise for child classes.